src/main/cpp/src/hive_hash.cu

```diff
@@ -210,11 +208,222 @@ class hive_device_row_hasher {
       return this->hash_functor.template operator()<T>(col, row_index);
     }
 
+    /**
+     * @brief A structure to keep track of the computation for nested types.
+     */
+    struct col_stack_frame {
+     private:
+      cudf::column_device_view _element;  // the column has only one row
```

Comment:
The comment is wrong, since when we push a (nested) child column of a lists column then it may have many rows. In addition, `element` should mean the row, so we actually should not rename this in the last time.
```suggestion
      cudf::column_device_view _column;
```

Reply:
For the `list column`, I only push one element of `child_col` each time.

Reply:
No, for lists column with nested child, you push that unique child column only once.